src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -105,9 +119,11 @@ public record VectorSearchConfig(
             @JsonProperty("service")
             VectorizeConfig vectorizeConfig) {
 
-      public VectorSearchConfig(Integer dimension, String metric, VectorizeConfig vectorizeConfig) {
+      public VectorSearchConfig(
+          Integer dimension, String metric, String sourceModel, VectorizeConfig vectorizeConfig) {
         this.dimension = dimension;
-        this.metric = metric == null ? "cosine" : metric;
```

Comment:
Is this change intentional wrt metric? (I assume so but since there is no comment thought I'd ask)

Reply:
Yes. The `metric` somehow depends on the `sourceModel` now. If the `sourceModel` is not `other` and the user doesn't give the value to `metric`. We need to assign `dot_product` to `metric` (I handle it in `CreateCollectionCommandResolver`). If we assign `cosine` here, we can't know whether the `cosine` is given by the user or by our default value.